From e87d4e0a93a7831c48af57a10e7ef71d4af32c7f Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Fri, 24 Jul 2026 04:20:46 -0400
Subject: [PATCH 231/268] atiadxx: More stubs for native AGS.

These will be implemented later (At least display modes will have to be populated at a minimum)
---
 dlls/atiadlxx/atiadlxx.spec   |  8 +++----
 dlls/atiadlxx/atiadlxx_main.c | 43 +++++++++++++++++++++++++++++++++++
 2 files changed, 47 insertions(+), 4 deletions(-)

diff --git a/dlls/atiadlxx/atiadlxx.spec b/dlls/atiadlxx/atiadlxx.spec
index 73dcfff2c1d..b2473bc1a22 100644
--- a/dlls/atiadlxx/atiadlxx.spec
+++ b/dlls/atiadlxx/atiadlxx.spec
@@ -281,7 +281,7 @@
 @ stub ADL2_Display_FormatsOverride_Set
 @ stub ADL2_Display_FreeSyncState_Get
 @ stub ADL2_Display_FreeSyncState_Set
-@ stub ADL2_Display_FreeSync_Cap
+@ cdecl ADL2_Display_FreeSync_Cap(ptr long long ptr)
 @ stub ADL2_Display_GamutMapping_Get
 @ stub ADL2_Display_GamutMapping_Reset
 @ stub ADL2_Display_GamutMapping_Set
@@ -315,7 +315,7 @@
 @ stub ADL2_Display_ModeTimingOverride_Delete
 @ stub ADL2_Display_ModeTimingOverride_Get
 @ stub ADL2_Display_ModeTimingOverride_Set
-@ stub ADL2_Display_Modes_Get
+@ cdecl ADL2_Display_Modes_Get(ptr long long ptr ptr)
 @ stub ADL2_Display_Modes_Set
 @ stub ADL2_Display_Modes_X2_Get
 @ stub ADL2_Display_MonitorPowerState_Set
@@ -865,7 +865,7 @@
 @ stub ADL_Display_FormatsOverride_Set
 @ stub ADL_Display_FreeSyncState_Get
 @ stub ADL_Display_FreeSyncState_Set
-@ stub ADL_Display_FreeSync_Cap
+@ cdecl ADL_Display_FreeSync_Cap(long long ptr)
 @ stub ADL_Display_GamutMapping_Get
 @ stub ADL_Display_GamutMapping_Reset
 @ stub ADL_Display_GamutMapping_Set
@@ -891,7 +891,7 @@
 @ stub ADL_Display_ModeTimingOverride_Delete
 @ stub ADL_Display_ModeTimingOverride_Get
 @ stub ADL_Display_ModeTimingOverride_Set
-@ stub ADL_Display_Modes_Get
+@ cdecl ADL_Display_Modes_Get(long long ptr ptr)
 @ stub ADL_Display_Modes_Set
 @ stub ADL_Display_MonitorPowerState_Set
 @ stub ADL_Display_NativeAUXChannel_Access
diff --git a/dlls/atiadlxx/atiadlxx_main.c b/dlls/atiadlxx/atiadlxx_main.c
index c5dbfb533a3..e83a5c1a96e 100644
--- a/dlls/atiadlxx/atiadlxx_main.c
+++ b/dlls/atiadlxx/atiadlxx_main.c
@@ -375,6 +375,21 @@ typedef struct ADLAdapterODClockInfo
     ADLODClockSetting sEngineClock;
 } ADLAdapterODClockInfo;
 
+typedef struct ADLFreeSyncCap
+{
+    /// FreeSync capability flags. \ref define_freesync_caps
+    int iCaps;
+    /// Reports minimum FreeSync refresh rate supported by the display in micro hertz
+    int iMinRefreshRateInMicroHz;
+    /// Reports maximum FreeSync refresh rate supported by the display in micro hertz
+    int iMaxRefreshRateInMicroHz;
+    /// Index of FreeSync Label to use:  ADL_FREESYNC_LABEL_*
+    unsigned char ucLabelIndex;
+    /// Reserved
+    char cReserved[3];
+    int iReserved[4];
+} ADLFreeSyncCap;
+
 typedef unsigned int ADL_D3DKMT_HANDLE;
 
 static const ADLVersionsInfo version = {
@@ -1300,6 +1315,34 @@ int CDECL ADL_Display_DisplayMapConfig_Get(int adapter_index, int *display_map_c
             display_target_count, display_targets, options);
 }
 
+int CDECL ADL2_Display_Modes_Get(ADL_CONTEXT_HANDLE ctx, int adapter_index, int display_index,
+                                 int *num_modes, ADLMode **modes)
+{
+    TRACE("ctx %p adapter_index %d display_index %d num_modes %p modes %p", ctx,
+          adapter_index, display_index, num_modes, modes);
+
+    /* FIXME: */
+
+    return ADL_ERR;
+}
+
+int CDECL ADL_Display_Modes_Get(int adapter_index, int display_index, int *num_modes, ADLMode **modes)
+{
+    return ADL2_Display_Modes_Get(default_ctx, adapter_index, display_index, num_modes, modes);
+}
+
+int CDECL ADL2_Display_FreeSync_Cap(ADL_CONTEXT_HANDLE ctx, int adapter_index, int display_index, ADLFreeSyncCap *caps)
+{
+    FIXME("ctx %p adapter_index %d display_index %d caps %p stub!\n", ctx, adapter_index, display_index, caps);
+
+    return ADL_ERR;
+}
+
+int CDECL ADL_Display_FreeSync_Cap(int adapter_index, int display_index, ADLFreeSyncCap *caps)
+{
+    return ADL2_Display_FreeSync_Cap(default_ctx, adapter_index, display_index, caps);
+}
+
 int CDECL ADL_Display_MVPUStatus_Get(int adapter_index, ADLMVPUStatus *mvpu_status)
 {
     TRACE("adapter_index %d, mvpu_status %p.\n", adapter_index, mvpu_status);
-- 
2.55.0

